Modernize perf dashboard service scripts to python3.

dewei-zhu opened the pull request:
#### 245fe23bcd36ed625f0c46ce225f407cf3a0bd3f
<pre>
Modernize perf dashboard service scripts to python3.
<a href="https://rdar.apple.com/117771755">rdar://117771755</a>

Reviewed by Jonathan Bedard.

Convert all perf dashboard service scripts to python3.

* Websites/perf.webkit.org/tools/pull-os-versions.py:
(main):
(OSBuildFetcher._fetch_available_builds):
(OSBuildFetcher.fetch_and_report_new_builds):
(available_builds_from_command):
(fetch_available_builds):
* Websites/perf.webkit.org/tools/sync-commits.py:
(main):
(Repository.fetch_commits_and_submit):
(Repository.fetch_revision_from_dasbhoard):
(SVNRepository.fetch_next_commit):
(SVNRepository._resolve_author_name):
(GitRepository._fetch_all_hashes):
(GitRepository._run_git_command):
* Websites/perf.webkit.org/tools/util.py:
(submit_commits):
(setup_auth):

Canonical link: <a href="https://commits.webkit.org/270155@main">https://commits.webkit.org/270155@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ca74c8e1dd35c7344aa40e5ec3b3dcafa70b7e6b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24696 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/3241 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25950 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26814 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22677 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/24964 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/4912 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/678 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/23035 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24940 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/2291 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/21325 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27397 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/2006 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/22256 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/28420 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22521 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22597 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26225 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1937 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/254 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/3237 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5917 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/2391 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/2298 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard reviewed (approved):
This looks good, although it's worth noting that this change doesn't just make these scripts Python 3 compatible, it breaks these scripts in Python 2. Given how we use these scripts, I believe that's acceptable, but I want to make sure that's been considered.

dewei-zhu committed:
Modernize perf dashboard service scripts to python3.
rdar://117771755

Reviewed by Jonathan Bedard.

Convert all perf dashboard service scripts to python3.

* Websites/perf.webkit.org/tools/pull-os-versions.py:
(main):
(OSBuildFetcher._fetch_available_builds):
(OSBuildFetcher.fetch_and_report_new_builds):
(available_builds_from_command):
(fetch_available_builds):
* Websites/perf.webkit.org/tools/sync-commits.py:
(main):
(Repository.fetch_commits_and_submit):
(Repository.fetch_revision_from_dasbhoard):
(SVNRepository.fetch_next_commit):
(SVNRepository._resolve_author_name):
(GitRepository._fetch_all_hashes):
(GitRepository._run_git_command):
* Websites/perf.webkit.org/tools/util.py:
(submit_commits):
(setup_auth):

Canonical link: https://commits.webkit.org/270155@main
